packages/plugin-http-cache/src/index.js

```diff
@@ -0,0 +1,149 @@
+import { bindWidgetToFunctions, hookMethod } from '@merkur/core';
+import { setDefaultConfig, copyResponse } from '@merkur/plugin-http-client';
+
+import CacheEntry from './CacheEntry';
+
+const DEV = 'development';
+const ENV =
+  typeof process !== 'undefined' && process && process.env
+    ? process.env.NODE_ENV
+    : DEV;
+
+export function getCacheKey({ method, url, body, query }) {
+  const data = ~['GET', 'HEAD'].indexOf(method) ? query : body;
+  let dataQuery = '';
+
+  if (data) {
+    try {
+      dataQuery = JSON.stringify(data).replace(/<\/script/gi, '<\\/script');
+    } catch (error) {
+      console.warn('The provided data does not have valid JSON format', data);
+    }
+  }
+  return `${method}:${url}?${dataQuery}`;
+}
+
+export function httpCachePlugin() {
+  return {
+    async setup(widget) {
+      if (ENV === DEV && !widget.$in.httpClient) {
+        throw new Error(
+          'You must setup plugin @merkur/plugin-http-client before @merkur/plugin-http-cache'
+        );
+      }
+
+      widget = {
+        ...httpCacheAPI(),
+        ...widget,
+      };
+
+      widget.$in.httpClient.cache = new Map();
+
+      return widget;
+    },
+    async create(widget) {
+      bindWidgetToFunctions(widget, widget.httpCache);
+
+      if (widget.$httpCache) {
+        widget.httpCache.deserialize(widget.$httpCache);
+      }
+
+      setDefaultConfig(widget, {
+        useCache: true,
+        ttl: 60000,
+      });
+
+      hookMethod(widget, 'info', infoHook);
```

Comment:
I dont know if you always want it by default. Because you have already all data in state.